internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubContext.java

```diff
@@ -0,0 +1,76 @@
+package com.linkedin.venice.pubsub;
+
+import com.linkedin.venice.pubsub.manager.TopicManager;
+import com.linkedin.venice.pubsub.manager.TopicManagerRepository;
+
+
+/**
+ * {@code PubSubContext} is a container class that holds all the core components required
+ * for managing PubSub infrastructure, including topic managers, position registry,
+ * position deserializer, and topic repository.
+ */
+public class PubSubContext {
+  private final TopicManagerRepository topicManagerRepository;
```

Comment:
Is it possible, or does it make sense, that we need only a subset of these four components in a PubSubContext ?

Reply:
I’ve included only the components relevant to this context. TopicManagerRepository and PubSubTopicRepository were already being propagated, so I consolidated them under PubSubContext to avoid passing them individually.

Reply:
I added a builder since we’ll be adding more dependencies later, and this avoids repeatedly changing the constructor.